### PR TITLE
Meltdown Scaling

### PR DESCRIPTION
**Pull request by dsmith328**
## About The Pull Request
Changes various facets from scaling based on time to scale off meltdowns and abnormality quantity. Affected are Abnormality Arrivals (Scaling with Players, Min 2), Agent spawn stats (Every 1/6th of the facility filled is buffed), and Zayin and Teth removal from the queue. Also added is a randomized Pizza reward for hitting Energy Goal, because it's nice. Alongside these changes, at least 1 abno is now forced to spawn at the round start after the usual 3 minutes of the world being up, but this excludes the 3 stat-check Teths and We Can Change anything. A second abno follows the moment the first meltdown occurs.

## Why It's Good For The Game
It's been requested by quite a few people and I think it would help smooth the game out overall, especially on rounds that get "abandoned" in the middle of the night (Full facility but no ego). I know you (Egor) want the Ordeals to still be able to disadvantage people, I think, so I would still be down for making Ordeals turn the auto-fire timer back on for the duration of an ordeal if you wanted. That way Ordeals actively speed up the game if not handled properly (or timely).

## Changelog
:cl:
add: Pizza Time
tweak: Changed Abno Arrival to be based on Meltdowns.
balance: Changed Agent late-join stats to be based off how full the facility is, instead of time.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/abnormality_queue.dm [hunk 1]
```diff
@@ -1,7 +1,7 @@
 SUBSYSTEM_DEF(abnormality_queue)
 	name = "Abnormality Queue"
 	flags = SS_KEEP_TIMING | SS_BACKGROUND
-	wait = 5 MINUTES
+	wait = INFINITY
```
Comment (EgorDinamit) [inline code/controllers/subsystem/abnormality_queue.dm:4]: If you are going to do this, then just make it not fire at all.

File: code/controllers/subsystem/lobotomy_corp.dm [hunk 2]
```diff
@@ -121,6 +134,10 @@ SUBSYSTEM_DEF(lobotomy_corp)
 	if(ran_ordeal)
 		return
 	InitiateMeltdown(qliphoth_meltdown_amount, FALSE)
+	if(abno_wait <= qliphoth_state)
+		SSabnormality_queue.fire() // Fired AFTER the meltdown occurs, so there's no situation where players are screwed because they were prepping for CENSORED and he arrives and instant meltdowns.
+		abno_wait_cooldown = round(clamp(1+(player_ratio*player_count), 1, INFINITY)) // Actively dynamic with the living players.
```
Comment (EgorDinamit) [inline code/controllers/subsystem/lobotomy_corp.dm:139]: Why is it scaling up with pop..?

File: code/controllers/subsystem/lobotomy_corp.dm [hunk 3]
```diff
@@ -165,7 +182,7 @@ SUBSYSTEM_DEF(lobotomy_corp)
 		return FALSE
 	next_ordeal = pick(available_ordeals)
 	all_ordeals[next_ordeal_level] -= next_ordeal
-	next_ordeal_time = qliphoth_state + (next_ordeal_level * 2) + rand(1,3)
+	next_ordeal_time = qliphoth_state*1.5 + (next_ordeal_level * 2) + rand(1,3) // Increased
```
Comment (EgorDinamit) [inline code/controllers/subsystem/lobotomy_corp.dm:185]: Why?

File: code/modules/jobs/job_types/agent.dm [hunk 4]
```diff
@@ -74,15 +74,17 @@
 		to_chat(M, "<b>You have not been assigned to any department.</b>")
 
 	var/set_attribute = normal_attribute_level
-	if(world.time >= 75 MINUTES) // Full facility expected
+	var/interval = SSabnormality_queue.spawned_abnos
+	var/max = GLOB.abnormality_room_spawners.len
```
Comment (EgorDinamit) [inline code/modules/jobs/job_types/agent.dm:78]: You do realize that the spawners delete and therefore get removed from the list when abnormalities spawn, right?

File: code/controllers/subsystem/lobotomy_corp.dm (hunk 2 above)
Comment (dsmith328) [reply code/controllers/subsystem/lobotomy_corp.dm:139]: When there's a lot of people 2 meltdowns or even 1 can be REALLY LOW and abnos could come in REALLY FAST. I wanna give people a chance to do shit.

File: code/controllers/subsystem/lobotomy_corp.dm (hunk 3 above)
Comment (dsmith328) [reply code/controllers/subsystem/lobotomy_corp.dm:185]: While testing it ordeals came WAY too fast, this is something I'm testing to see how people feel about it. Last time they simply didn't get enough abnos fast enough to be reasonable.

File: code/modules/jobs/job_types/agent.dm (hunk 4 above)
Comment (dsmith328) [reply code/modules/jobs/job_types/agent.dm:78]: I didn't but that's fucking amazing to know, I can really easily fix that by referencing the abno queue's "Room_Start" variable which is only set at round start.

File: code/controllers/subsystem/abnormality_queue.dm (hunk 1 above)
Comment (dsmith328) [reply code/controllers/subsystem/abnormality_queue.dm:4]: Yeah, temp fix that I'll change.

**Comment by dsmith328** [comment]
Closing this for the time being as to not take up space. I may look at this further or even just use this as a learning experience.

**Pull request closed without merging**